### ISSUE TITLE
Builder thinks he will get more items delivered than requested from crafter

### ISSUE THREAD

**Issue by MatthiasMann** [ldtteam/minecolonies#5395]
Version: minecolonies-0.13.37-ALPHA-universal (but was also in 0.13.34)

### Expected behavior
Builders gets the items delivered he requests and then continue working.

### Actual behaviour
At some point the request for things to get crafted changes to this:
![2020-07-17_22 14 10](https://user-images.githubusercontent.com/6777029/87828180-b0724480-c87c-11ea-8c0a-1b780bfd5cab.png)
and once he gets the items delivered (which is the original number) he continues to wait for more - until you cancel the request.

### Steps to reproduce the problem
1. Have a sawmill setup to craft planks, stairs, signs, doors etc
2. Have the builder request crafting of one or more of these items
3. At some point the crafting / delivering amount starts to go up
4. ... Profit

### Logs: Latest.log and crashlog at https://gist.github.com/
No messages in either latest.log or debug.log from Minecolonies since world load (this issue can also happen later not only after loading a world).

**Comment by MatthiasMann** [comment]
Now it also happened to the stone mason crafting items:
![2020-07-17_22 56 54](https://user-images.githubusercontent.com/6777029/87830329-1791f800-c881-11ea-80f9-2eac6067d49d.png)
![2020-07-17_22 56 59](https://user-images.githubusercontent.com/6777029/87830334-195bbb80-c881-11ea-81a7-d33839be2030.png)

**Comment by MatthiasMann** [comment]
Still happens in 0.13.56

**Comment by Raycoms** [comment]
Can you see if this still works with the latest?

**Comment by LuisFer666** [comment]
Still happens in minecolonies-0.13.105-ALPHA-universal

<a href="https://imgur.com/yjCcdIL"><img src="https://i.imgur.com/yjCcdIL.png" title="source: imgur.com" /></a>

**Comment by MatthiasMann** [comment]
Still happens in 0.13.120:
![2020-08-10_19 06 07](https://user-images.githubusercontent.com/6777029/89810060-c18d3900-db3c-11ea-9251-0e75fbbf4762.png)

**Comment by kpiatkowski** [comment]
Got this on 0.13.144 too (and earlier), happened when delivery man responsible for fulfilling that request vanished.

**Comment by MatthiasMann** [comment]
I think I now have an idea what happens with the crafters and the delivery counts always going up:
I had the builder requesting 96 stone from my stone smelter - but the delivery guy brought him 128 stone, which cause the request to not complete and instead another delivery of stone was made - I'm on 1.15.2 and 0.13.247
I think this is similar to when you (eg the player) put more items as requested into the builder's inventory the request is also not fulfilled and stays active

**Comment by ravenbuilder934** [comment]
This happens to me too-at one point I had 30 iron helmets in my warehouse! Not just the builder, at least for me.

**Comment by justinbey** [comment]
I've only ever seen it happen with my blacksmith but they will just make tools and armor over and over again and waste resources. I look in the chests in the warehouse and I have tons of tools and armor just sitting there.

**Comment by bugSandwich** [comment]
This also happens in the latest 1.15.2 beta. I will often find that someone has dozens of copies of the requested item their hut's inventory, but they still hasn't closed their request, or the builder will have received everything but will still request building materials.

**Comment by Talyda** [comment]
.264 or .250 BugSandwich?
Either way, best to update to the latest alpha

**Comment by Aduis1210** [comment]
I had 20 beds made due to this.

**Comment by MatthiasMann** [comment]
Yesterday it also happened when the stone smelter requested wooden planks as fuel from my saw mill. Each time he got planks delivered the amount of requested increased. Using 0.13.349-alpha.

**Comment by Aduis1210** [comment]
Ive not checked the stone mason. But id imagine it be the same.

**Comment by DrFlash99** [comment]
Just tested this on 0.13.350 and it still occurs. Heres what I did to reproduce the problem.

my builder had a task pending to upgrade the tavern, one of the items required was 13 vertical barrel.
timeline:
1. builder requests 1 vertical barrel
2. mats items get delivered to sawmill 
3. 1 barrel gets made
4. Builder request times out -> go to step 1.
- Above happened 6 times before the items got picked up by a delivery npc, who stored them in the warehouse
5. Builder requests 1 vertical barrel
6. 6 barrels get send from warehouse to builder.

**Comment by ravenbuilder934** [comment]
In my opinion this is the most annoying bug that still exists. I have like 10 stacks of birch buttons and I had almost an entire rack full of cobblestone slabs (until I turned them back into cobble)!

**Comment by Raycoms** [comment]
Does this still happen in the latest? Can someone test?

**Comment by ravenbuilder934** [comment]
It happens on the latest minecolonies official version (v2.9), but that's over a week out of date.

**Comment by Aduis1210** [comment]
Checking

**Comment by Aduis1210** [comment]
nope my builder is still requesting more bookcases to be made then needed

**Comment by Raycoms** [comment]
Are you on the latest alpha that is? the one we released today?

**Comment by Aduis1210** [comment]
I am yes Ray6

**Comment by Aduis1210** [comment]
0.13.372 im on

**Comment by Raycoms** [comment]
kk

**Comment by Aduis1210** [comment]
Just noticed, The Miner does the same when requesting items for the tunnels.  My miner had 7 stacks of stairs but still wanted 100 more via the request system.

**Comment by ravenbuilder934** [comment]
I think it can happen to anyone, my blacksmith has made loads of armor for my guards and also lots of axes for forester

**Comment by Aduis1210** [comment]
https://paste.ee/p/3PnJC - this error shows when i try to cancel a Request system request
It doesn't cancel at all

**Comment by ravenbuilder934** [comment]
> https://paste.ee/p/3PnJC - this error shows when i try to cancel a Request system request
> It doesn't cancel at all

Probably unrelated, open a separate issue

**Comment by Malorn44** [comment]
I don't think it is the same issue. Or if it is I find no fix. I try to cancel the job, reset the request system, etc... Do you have any additional suggestions for things I can try then? I've done just about everything https://wiki.minecolonies.ldtteam.com/source/misc/troubleshooting <- here

**Comment by ravenbuilder934** [comment]
Don't think what is the same issue?

**Comment by Nikademus** [comment]
Is this the same issue?  I requested one dye, when the courier delivered the flower to the dye hut, they sent 7 dyes back to me for delivery.
![2020-10-24_11 49 36](https://user-images.githubusercontent.com/1380028/97086192-d469c280-15ef-11eb-9507-b31933101d83.png)

**Comment by Raycoms** [comment]
It's all related to the "wrong quantities" though

**Comment by Nikademus** [comment]
Followup, it just suddenly went back to requesting a flower delivery, I never got the dyes.  I went to the dye hut and the seven dyes were sitting there.
![2020-10-24_18 53 54](https://user-images.githubusercontent.com/1380028/97095200-7a3a2300-162a-11eb-92ce-6c3e7c723e1d.png)

**Comment by perecnicek** [comment]
I have this issue in version minecolonies-0.13.379-ALPHA-universal with my blacksmith i have warehouse full of tools like stone axe shiedls shovels

**Comment by xiaobbl** [comment]
now my cowboy get this:(


<a href="https://sm.ms/image/aK6rGbZ5uwvW2nE" target="_blank"><img src="https://i.loli.net/2020/10/31/aK6rGbZ5uwvW2nE.png" ></a>
running the latest alpha(0.13.394)

**Comment by xiaobbl** [comment]
canceling the request and let citizens make again can make it better(for a peries of time)

**Comment by Knaxia** [comment]
I can confirm this can also happen with any request from the system, where whoever is crafting will constantly request material and craft a ton.

Here are the many occurrences that I had:
-Builders asking for something from stonemason or sawmill, randomly would keep requesting a certain item and you would see the "delivery of 64" a few times when they only asked for a small amount
-Guards asking for a piece of armor
-I had a guard complain that he 1: needed new pants and 2: that he had no space left in his hut to store things. Upon verification, I canceled his request because he clearly had pants on him, and checking his hut, the entire thing was filled with leather pants, I had to destroy most of them and call a pickup for the last one remaining.
-I had another similar with iron chest plate. That was a lot of chest plates that once appeared in my warehouse after I had to do a colony request reset all command to fix another issue.
-I asked for 2 dyes from my Dyer, I ended up with 16 in my mailbox
-My miner kept asking for cobblestone oak double-crossed blocks, I noticed he kept getting some, he would drop them in one of his rack/chest but then seem to forget about them and request more. I had around 20 stacks before I noticed. I picked everything up, removed the chests and racks, replaced them and gave my miner all the items. He finally figured it out, placed them in storage and went on his day. The next pickup took ALL of the blocks and put them in the warehouse...

That still happens randomly, even in 13.390 and 13.396

**Comment by Jester1o1** [comment]
I was brought here after finding issue 
Farmer continuously requests hoe #5983

My Farmers stand by their hut block asking for hoes continuously.
minecolonies-0.13.359-BETA-universal.jar
structurize-0.13.94-ALPHA-universal.jar
Minecraft 1.15.2

Tried multiple of the troubleshooting attempts above. Nothing seems to work.
If you'd like logs I can always try to provide.

**Comment by xiaobbl** [comment]
has make sure that happened on 0.13.421

**Comment by timstoop** [comment]
I have this problem as well, but only when resources take several (in game) days to be delivered and are craftables. It looks to me like in this situation, at some point, a new request is created for the crafting and the inventory (either the local or the warehouse) are not checked if the items already exist.

I'm not sure if the new request is expected, but if that would check the inventories again if the items are available, that should solve the issue, I think.

**Comment by radoslawKnitter** [comment]
still in minecolonies-0.13.440-ALPHA-universal

**Comment by Noridin** [comment]
Cook requests endless amounts of oak planks for fuel in 0.13.450-ALPHA

**Comment by xiaobbl** [comment]
now it seems that the things that they dont need so much would not deliver to them.but of course their requests cant be finished and they have to request one more time.
